src/CMakeLists.txt

```diff
@@ -177,8 +167,42 @@ set(IIDM_SOURCES
     xml/XmlStreamWriter.cpp
 )
 
-add_library(iidm ${IIDM_SOURCES})
+# Shared library
+add_library(iidm SHARED ${IIDM_SOURCES})
+set_target_properties(iidm PROPERTIES SOVERSION ${IIDM_SOVERSION} VERSION ${IIDM_VERSION})
+target_include_directories(iidm SYSTEM PUBLIC ${Boost_INCLUDE_DIRS} ${LIBXML2_INCLUDE_DIR})
```

Comment:
I would use something like this $<TARGET_PROPERTY:Boost::boost,INTERFACE_INCLUDE_DIRECTORIES> and idem for libxml2

Reply:
Could you explain the difference?
It seems I don't have an export for LibXml2. It could be done only for Boost then.